I've been seeing this when running QuarkusTest

```
Error: Exception in thread "vert.x-internal-blocking-1" java.lang.IllegalAccessError: module java.base does not open java.lang to unnamed module @18adc20a; to use the thread-local-reset capability on Java 24 or later, use this JVM option: --add-opens java.base/java.lang=ALL-UNNAMED
	at org.jboss.threads.JDKSpecific$ThreadAccess.<clinit>(JDKSpecific.java:32)
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:13)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Error: Exception in thread "executor-thread-1" java.lang.NoClassDefFoundError: Could not initialize class org.jboss.threads.JDKSpecific$ThreadAccess
	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:13)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:1474)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.IllegalAccessError: module java.base does not open java.lang to unnamed module @18adc20a; to use the thread-local-reset capability on Java 24 or later, use this JVM option: --add-opens java.base/java.lang=ALL-UNNAMED [in thread "vert.x-internal-blocking-1"]
	at org.jboss.threads.JDKSpecific$ThreadAccess.<clinit>(JDKSpecific.java:32)
	... 3 more
```

--- a/core/deployment/src/main/java/io/quarkus/deployment/images/ContainerImages.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/images/ContainerImages.java
@@ -1,6 +1,7 @@
 package io.quarkus.deployment.images;
 
 import io.quarkus.deployment.pkg.builditem.CompiledJavaVersionBuildItem;
+import io.quarkus.deployment.pkg.builditem.CompiledJavaVersionBuildItem.JavaVersion.Status;
 
 /**
  * This class is used to define the container images that are used by Quarkus.
@@ -33,12 +34,12 @@ public class ContainerImages {
     /**
      * Version used for more UBI9 Java images.
      */
-    public static final String UBI9_JAVA_VERSION = "1.23";
+    public static final String UBI9_JAVA_VERSION = "1.24";
 
     /**
      * Version uses for the native builder image.
      */
-    public static final String NATIVE_BUILDER_VERSION = "jdk-21";
+    public static final String NATIVE_BUILDER_VERSION = "jdk-25";
 
     // === Runtime images for containers (native)
 
@@ -89,6 +90,11 @@ public class ContainerImages {
     public static final String UBI9_JAVA_21_VERSION = UBI9_JAVA_VERSION;
     public static final String UBI9_JAVA_21 = UBI9_JAVA_21_IMAGE_NAME + ":" + UBI9_JAVA_21_VERSION;
 
+    // UBI 9 OpenJDK 25 Runtime - https://catalog.redhat.com/en/software/containers/ubi9/openjdk-25-runtime/69204990c46419100ce30a5b
+    public static final String UBI9_JAVA_25_IMAGE_NAME = "registry.access.redhat.com/ubi9/openjdk-25-runtime";
+    public static final String UBI9_JAVA_25_VERSION = UBI9_JAVA_VERSION;
+    public static final String UBI9_JAVA_25 = UBI9_JAVA_25_IMAGE_NAME + ":" + UBI9_JAVA_25_VERSION;
+
     // === Source To Image images
 
     // UBI 8 Quarkus Binary Source To Image - https://quay.io/repository/quarkus/ubi-quarkus-native-binary-s2i?tab=tags
@@ -118,6 +124,11 @@ public class ContainerImages {
     public static final String S2I_JAVA_21_VERSION = UBI9_JAVA_VERSION;
     public static final String S2I_JAVA_21 = S2I_JAVA_21_IMAGE_NAME + ":" + S2I_JAVA_21_VERSION;
 
+    // Java Source To Image - https://catalog.redhat.com/en/software/containers/ubi9/openjdk-25/69204893fc0d23c633bf5c29
+    public static final String S2I_JAVA_25_IMAGE_NAME = "registry.access.redhat.com/ubi9/openjdk-25";
+    public static final String S2I_JAVA_25_VERSION = UBI9_JAVA_VERSION;
+    public static final String S2I_JAVA_25 = S2I_JAVA_25_IMAGE_NAME + ":" + S2I_JAVA_25_VERSION;
+
     // === Native Builder images
 
     // Mandrel Builder Image - https://quay.io/repository/quarkus/ubi-quarkus-mandrel-builder-image?tab=tags
@@ -141,11 +152,14 @@ public class ContainerImages {
     public static final String UBI9_GRAALVM_BUILDER = UBI9_GRAALVM_BUILDER_IMAGE_NAME + ":" + UBI9_GRAALVM_BUILDER_VERSION;
 
     public static String getDefaultJvmImage(CompiledJavaVersionBuildItem.JavaVersion version) {
-        switch (version.isJava21OrHigher()) {
-            case TRUE:
-                return UBI9_JAVA_21;
-            default:
-                return UBI9_JAVA_17;
+        if (version.isJava25OrHigher() == Status.TRUE) {
+            return UBI9_JAVA_25;
         }
+
+        if (version.isJava21OrHigher() == Status.TRUE) {
+            return UBI9_JAVA_21;
+        }
+
+        return UBI9_JAVA_17;
     }
 }

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/builditem/CompiledJavaVersionBuildItem.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/builditem/CompiledJavaVersionBuildItem.java
@@ -29,6 +29,8 @@ public final class CompiledJavaVersionBuildItem extends SimpleBuildItem {
 
     public interface JavaVersion {
 
+        Status isJava25OrHigher();
+
         Status isJava21OrHigher();
 
         Status isJava19OrHigher();
@@ -42,6 +44,11 @@ public final class CompiledJavaVersionBuildItem extends SimpleBuildItem {
         final class Unknown implements JavaVersion {
 
             Unknown() {
+            }
+
+            @Override
+            public Status isJava25OrHigher() {
+                return Status.UNKNOWN;
             }
 
             @Override
@@ -59,6 +66,7 @@ public final class CompiledJavaVersionBuildItem extends SimpleBuildItem {
 
             private static final int JAVA_19_MAJOR = 63;
             private static final int JAVA_21_MAJOR = 65;
+            private static final int JAVA_25_MAJOR = 69;
 
             private final int determinedMajor;
 
@@ -74,6 +82,11 @@ public final class CompiledJavaVersionBuildItem extends SimpleBuildItem {
             @Override
             public Status isJava21OrHigher() {
                 return higherOrEqualStatus(JAVA_21_MAJOR);
+            }
+
+            @Override
+            public Status isJava25OrHigher() {
+                return higherOrEqualStatus(JAVA_25_MAJOR);
             }
 
             private Status higherOrEqualStatus(int javaMajor) {

--- a/core/deployment/src/main/java/io/quarkus/deployment/steps/ApplyNativeImageAgentConfigStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/steps/ApplyNativeImageAgentConfigStep.java
@@ -40,16 +40,21 @@ public class ApplyNativeImageAgentConfigStep {
             if (!targetPath.toFile().exists()) {
                 targetPath.toFile().mkdirs();
             }
-            Files.copy(basePath.resolve("reflect-config.json"), targetPath.resolve("reflect-config.json"),
-                    StandardCopyOption.REPLACE_EXISTING);
-            Files.copy(basePath.resolve("serialization-config.json"), targetPath.resolve("serialization-config.json"),
-                    StandardCopyOption.REPLACE_EXISTING);
-            Files.copy(basePath.resolve("jni-config.json"), targetPath.resolve("jni-config.json"),
-                    StandardCopyOption.REPLACE_EXISTING);
-            Files.copy(basePath.resolve("proxy-config.json"), targetPath.resolve("proxy-config.json"),
-                    StandardCopyOption.REPLACE_EXISTING);
-            Files.copy(basePath.resolve("resource-config.json"), targetPath.resolve("resource-config.json"),
-                    StandardCopyOption.REPLACE_EXISTING);
+            if (Files.exists(basePath.resolve("reachability-metadata.json"))) {
+                Files.copy(basePath.resolve("reachability-metadata.json"), targetPath.resolve("reachability-metadata.json"),
+                        StandardCopyOption.REPLACE_EXISTING);
+            } else {
+                Files.copy(basePath.resolve("reflect-config.json"), targetPath.resolve("reflect-config.json"),
+                        StandardCopyOption.REPLACE_EXISTING);
+                Files.copy(basePath.resolve("serialization-config.json"), targetPath.resolve("serialization-config.json"),
+                        StandardCopyOption.REPLACE_EXISTING);
+                Files.copy(basePath.resolve("jni-config.json"), targetPath.resolve("jni-config.json"),
+                        StandardCopyOption.REPLACE_EXISTING);
+                Files.copy(basePath.resolve("proxy-config.json"), targetPath.resolve("proxy-config.json"),
+                        StandardCopyOption.REPLACE_EXISTING);
+                Files.copy(basePath.resolve("resource-config.json"), targetPath.resolve("resource-config.json"),
+                        StandardCopyOption.REPLACE_EXISTING);
+            }
 
             log.info("Applying native image agent generated files to current native executable build");
             nativeImageAgentConfigDirectoryProducer.produce(new NativeImageAgentConfigDirectoryBuildItem(targetDirName));

--- a/core/deployment/src/test/java/io/quarkus/deployment/pkg/NativeConfigTest.java
+++ b/core/deployment/src/test/java/io/quarkus/deployment/pkg/NativeConfigTest.java
@@ -9,20 +9,20 @@ class NativeConfigTest {
     @Test
     public void testBuilderImageProperlyDetected() {
         assertThat(createConfig("graalvm").builderImage().getEffectiveImage()).contains("ubi9-quarkus-graalvmce-builder-image")
-                .contains("jdk-21");
+                .contains("jdk-25");
         assertThat(createConfig("GraalVM").builderImage().getEffectiveImage()).contains("ubi9-quarkus-graalvmce-builder-image")
-                .contains("jdk-21");
+                .contains("jdk-25");
         assertThat(createConfig("GraalVM").builderImage().getEffectiveImage()).contains("ubi9-quarkus-graalvmce-builder-image")
-                .contains("jdk-21");
+                .contains("jdk-25");
         assertThat(createConfig("GRAALVM").builderImage().getEffectiveImage()).contains("ubi9-quarkus-graalvmce-builder-image")
-                .contains("jdk-21");
+                .contains("jdk-25");
 
         assertThat(createConfig("mandrel").builderImage().getEffectiveImage()).contains("ubi9-quarkus-mandrel-builder-image")
-                .contains("jdk-21");
+                .contains("jdk-25");
         assertThat(createConfig("Mandrel").builderImage().getEffectiveImage()).contains("ubi9-quarkus-mandrel-builder-image")
-                .contains("jdk-21");
+                .contains("jdk-25");
         assertThat(createConfig("MANDREL").builderImage().getEffectiveImage()).contains("ubi9-quarkus-mandrel-builder-image")
-                .contains("jdk-21");
+                .contains("jdk-25");
 
         assertThat(createConfig("aRandomString").builderImage().getEffectiveImage()).isEqualTo("aRandomString");
     }

--- a/devtools/maven/src/main/java/io/quarkus/maven/NativeImageAgentMojo.java
+++ b/devtools/maven/src/main/java/io/quarkus/maven/NativeImageAgentMojo.java
@@ -18,6 +18,8 @@ import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.ResolutionScope;
 
 import io.quarkus.bootstrap.json.Json;
+import io.quarkus.bootstrap.json.Json.JsonObjectBuilder;
+import io.quarkus.bootstrap.json.JsonArray;
 import io.quarkus.bootstrap.json.JsonMember;
 import io.quarkus.bootstrap.json.JsonObject;
 import io.quarkus.bootstrap.json.JsonReader;
@@ -59,7 +61,15 @@ public class NativeImageAgentMojo extends QuarkusBootstrapMojo {
                 }
                 getLog().debug("Native image agent config folder exits, copy and transform to " + targetPath);
                 final Path reflectConfigJsonPath = basePath.resolve("reflect-config.json");
-                if (reflectConfigJsonPath.toFile().exists()) {
+                final Path reachabilityMetadataJsonPath = basePath.resolve("reachability-metadata.json");
+                if (Files.exists(reachabilityMetadataJsonPath)) {
+                    // GraalVM/Mandrel 25+
+                    transformReachabilityMetadataJson(basePath, "reachability-metadata.json", targetPath);
+
+                    if (getLog().isInfoEnabled()) {
+                        getLog().info("Discovered native image agent generated files in " + targetPath);
+                    }
+                } else if (Files.exists(reflectConfigJsonPath)) {
                     Files.copy(reflectConfigJsonPath, targetPath.resolve("reflect-config.json"),
                             StandardCopyOption.REPLACE_EXISTING);
                     Files.copy(basePath.resolve("serialization-config.json"), targetPath.resolve("serialization-config.json"),
@@ -69,7 +79,7 @@ public class NativeImageAgentMojo extends QuarkusBootstrapMojo {
                     Files.copy(basePath.resolve("proxy-config.json"), targetPath.resolve("proxy-config.json"),
                             StandardCopyOption.REPLACE_EXISTING);
                     transformJsonObject(basePath, "resource-config.json", targetPath,
-                            JsonTransform.dropping(this::discardResource));
+                            JsonTransform.dropping(v -> discardResource("pattern", v)));
 
                     if (getLog().isInfoEnabled()) {
                         getLog().info("Discovered native image agent generated files in " + targetPath);
@@ -89,6 +99,25 @@ public class NativeImageAgentMojo extends QuarkusBootstrapMojo {
         }
     }
 
+    private void transformReachabilityMetadataJson(Path base, String name, Path target) throws IOException {
+        getLog().debug("Discarding resources from native image configuration that match the following regular expression: "
+                + resourceSkipPattern);
+        final String original = Files.readString(base.resolve(name));
+        final JsonObject jsonRead = JsonReader.of(original).read();
+        JsonArray resources = jsonRead.get("resources");
+        if (resources != null) {
+            final JsonObjectBuilder jsonBuilder = Json.object();
+            jsonBuilder.transform(jsonRead, JsonTransform.dropping(v -> discardResource("glob", v)));
+
+            try (BufferedWriter writer = new BufferedWriter(
+                    new FileWriter(target.resolve(name).toFile(), StandardCharsets.UTF_8))) {
+                jsonBuilder.appendTo(writer);
+            }
+        } else {
+            Files.copy(base.resolve(name), target.resolve(name), StandardCopyOption.REPLACE_EXISTING);
+        }
+    }
+
     private void transformJsonObject(Path base, String name, Path target, JsonTransform transform) throws IOException {
         getLog().debug("Discarding resources from native image configuration that match the following regular expression: "
                 + resourceSkipPattern);
@@ -103,10 +132,10 @@ public class NativeImageAgentMojo extends QuarkusBootstrapMojo {
         }
     }
 
-    private boolean discardResource(JsonValue value) {
+    private boolean discardResource(String attribute, JsonValue value) {
         if (value instanceof JsonMember) {
             final JsonMember member = (JsonMember) value;
-            if ("pattern".equals(member.attribute().value())) {
+            if (attribute.equals(member.attribute().value())) {
                 final JsonString memberValue = (JsonString) member.value();
                 final boolean discarded = resourceSkipPattern.matcher(memberValue.value()).find();
                 if (discarded) {

--- a/extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/ContainerImageOpenshiftConfig.java
+++ b/extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/ContainerImageOpenshiftConfig.java
@@ -29,6 +29,9 @@ public interface ContainerImageOpenshiftConfig {
     public static final String FALLBACK_NATIVE_BINARY_DIRECTORY = "/home/quarkus/";
 
     public static String getDefaultJvmImage(CompiledJavaVersionBuildItem.JavaVersion version) {
+        if (version.isJava25OrHigher() == CompiledJavaVersionBuildItem.JavaVersion.Status.TRUE) {
+            return ContainerImages.S2I_JAVA_25;
+        }
         if (version.isJava21OrHigher() == CompiledJavaVersionBuildItem.JavaVersion.Status.TRUE) {
             return ContainerImages.S2I_JAVA_21;
         }

--- a/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/JavaVersion.java
+++ b/independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/project/JavaVersion.java
@@ -66,11 +66,11 @@ public final class JavaVersion {
     }
 
     // ordering is important here, so let's keep them ordered
-    public static final SortedSet<Integer> JAVA_VERSIONS_LTS = new TreeSet<>(List.of(17, 21));
+    public static final SortedSet<Integer> JAVA_VERSIONS_LTS = new TreeSet<>(List.of(17, 21, 25));
     public static final int DEFAULT_JAVA_VERSION = 17;
     // we want to maximize the compatibility of extensions with the Quarkus ecosystem so let's stick to 17 by default
     public static final String DEFAULT_JAVA_VERSION_FOR_EXTENSION = "17";
-    public static final int MAX_LTS_SUPPORTED_BY_KOTLIN = 21;
+    public static final int MAX_LTS_SUPPORTED_BY_KOTLIN = 25;
     public static final String DETECT_JAVA_RUNTIME_VERSION = "<<detect java runtime version>>";
     public static final Pattern JAVA_VERSION_PATTERN = Pattern.compile("(\\d+)(?:\\..*)?");
 

--- a/independent-projects/tools/devtools-common/src/test/java/io/quarkus/devtools/project/JavaVersionTest.java
+++ b/independent-projects/tools/devtools-common/src/test/java/io/quarkus/devtools/project/JavaVersionTest.java
@@ -36,9 +36,20 @@ class JavaVersionTest {
     }
 
     @Test
+    public void givenJavaVersion26ShouldReturn25() {
+        assertEquals("25", computeJavaVersion(JAVA, "26.0.1"));
+    }
+
+    @Test
+    public void givenJavaVersion25ShouldReturn25() {
+        assertEquals("25", computeJavaVersion(JAVA, "25"));
+    }
+
+    @Test
     void shouldProperlyUseMinJavaVersion() {
-        assertThat(getCompatibleLTSVersions(new JavaVersion("17"))).containsExactly(17, 21);
-        assertThat(getCompatibleLTSVersions(new JavaVersion("21"))).containsExactly(21);
+        assertThat(getCompatibleLTSVersions(new JavaVersion("17"))).containsExactly(17, 21, 25);
+        assertThat(getCompatibleLTSVersions(new JavaVersion("21"))).containsExactly(21, 25);
+        assertThat(getCompatibleLTSVersions(new JavaVersion("25"))).containsExactly(25);
         assertThat(getCompatibleLTSVersions(new JavaVersion("100"))).isEmpty();
         assertThat(getCompatibleLTSVersions(JavaVersion.NA)).isEqualTo(JAVA_VERSIONS_LTS);
     }
@@ -55,6 +66,8 @@ class JavaVersionTest {
         assertEquals(17, determineBestJavaLtsVersion(18));
         assertEquals(21, determineBestJavaLtsVersion(21));
         assertEquals(21, determineBestJavaLtsVersion(22));
+        assertEquals(25, determineBestJavaLtsVersion(25));
+        assertEquals(25, determineBestJavaLtsVersion(26));
     }
 
     @Test

